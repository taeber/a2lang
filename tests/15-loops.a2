asm {
	ORG $2000
}

use [
    COUT : sub <- [ch: char @ A]   @ $FDED
    DOS  : sub @ $3D0
    HOME : sub @ $FC58,
    INIT : sub @ $FB2F,  ; AKA TEXT
    RDKEY: sub -> [key: byte @ A]  @ $FD0C
]

INIT()
HOME()

var key: byte

let CR = $8D

key := CR

while 0 == 0 {
    while 1 <> 0 {
        key := RDKEY()
        if key == `A {
            repeat
        }
        COUT()
        if key == `Q {
            stop
        }
    }
    stop
}

DOS()
