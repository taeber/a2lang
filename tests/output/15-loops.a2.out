	ORG $2000
COUT	EQU $FDED
DOS	EQU $3D0
HOME	EQU $FC58
INIT	EQU $FB2F
RDKEY	EQU $FD0C
	JSR INIT
	JSR HOME
CR	EQU $8D
* COPYBB key #CR
	LDA #CR
	STA key
* BRA A2_1
A2_0	CLV
	BVC A2_1
* IFNE #$01 #$00
*   A2_3 A2_4
A2_1	LDA #$01
	BNE A2_3
	JMP A2_4
A2_3	JSR RDKEY
* COPYBB key @A
	STA key
* IFEQ key #"A"
*   A2_6 A2_7
A2_5	LDA key
	CMP #"A"
	BEQ A2_6
	JMP A2_7
* REPEAT
A2_6	JMP A2_1
A2_7	JSR COUT
* IFEQ key #"Q"
*   A2_9 A2_10
A2_8	LDA key
	CMP #"Q"
	BEQ A2_9
	JMP A2_10
* STOP
A2_9	JMP A2_4
A2_10	JMP A2_1
* STOP
A2_4	JMP A2_2
	JMP A2_0
A2_2	JSR DOS
key	HEX 00
